src/Core/Table/DataProvider/RequestSqlTableDataProvider.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Table\DataProvider;
+
+use PrestaShopBundle\Entity\Repository\RequestSqlRepository;
+use PrestaShopBundle\Service\Hook\HookDispatcher;
+
+/**
+ * Class RequestSqlTableDataProvider provides data for Request SQL table
+ */
+final class RequestSqlTableDataProvider implements TableDataProviderInterface
+{
+    /**
+     * @var HookDispatcher
+     */
+    private $dispatcher;
+
+    /**
+     * @var RequestSqlRepository
+     */
+    private $repository;
+
+    /**
+     * @param HookDispatcher        $dispatcher
+     * @param RequestSqlRepository  $repository
+     */
+    public function __construct(
+        HookDispatcher $dispatcher,
+        RequestSqlRepository $repository
+    ) {
+        $this->dispatcher = $dispatcher;
+        $this->repository = $repository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRows(array $filters)
+    {
+        $requestSqlsQuery = $this->repository->getFindByFiltersQuery($filters);
+
+        // allow developers to modify Request SQL db query
+        $this->dispatcher->dispatchForParameters('modifyRequestSqlTableQuery', [
```

Comment:
I'm totally in favor of something like that :)

Reply:
But it seems they can't be modified by our module developers. We should send them as reference.

Reply:
aren't object passed as a reference? to be honest i didnt try it yet, so if it doesnt work we can pass it with reference.